app/src/androidTest/java/com/example/evently/ViewNotificationsTest.java

```diff
@@ -286,145 +327,110 @@ public void expectNotification_loser() throws InterruptedException {
+                .cancelled()
+                .contains(mockAccount.email()));
 
         // Confirms notification is seen.
-        notifications = notificationDB.fetchEventNotifications(invite.eventId()).await();
-        for (Notification n : notifications) {
-            if (n.hasSeen(mockAccount.email())) {
-                seenCount--;
-            }
-        }
-        assertEquals(-1, seenCount);
-        Thread.sleep(2000);
-    }
-
-    @Test
-    public void noExpectNotification_otherEvent() throws InterruptedException {
-        // Notifications from non-participated event should NOT show up.
-        Notification nonParticipatedEvent1 = templateNotification(8, Notification.Channel.Winners);
-        Notification nonParticipatedEvent2 = templateNotification(8, Notification.Channel.All);
-        Notification nonParticipatedEvent3 = templateNotification(0, Notification.Channel.All);
-
-        // Checks that no notifications for non-enrolled events show up.
-        Thread.sleep(2000);
-        onView(withText(nonParticipatedEvent1.title())).check(doesNotExist());
-        onView(withText(nonParticipatedEvent2.title())).check(doesNotExist());
-        onView(withText(nonParticipatedEvent3.title())).check(doesNotExist());
+        assertTrue(notificationDB
+                .fetchNotification(invite.id())
+                .await()
+                .get()
+                .hasSeen(mockAccount.email()));
     }
 
-    @Test
-    public void noExpectNotification_otherChannel() throws InterruptedException {
-        // checks that no notifications are shown when the event is participated in, but enterant is
-        // in a different channel:
-        Notification notification1 = templateNotification(2, Notification.Channel.Losers);
-        Notification notification2 = templateNotification(3, Notification.Channel.Losers);
-        Notification notification3 = templateNotification(7, Notification.Channel.Losers);
-        Notification notification4 = templateNotification(1, Notification.Channel.Winners);
-
-        // Confirms that none of the notifications whee found.
-        Thread.sleep(2000);
-        onView(withText(notification1.title())).check(doesNotExist());
-        onView(withText(notification2.title())).check(doesNotExist());
-        onView(withText(notification3.title())).check(doesNotExist());
-        onView(withText(notification4.title())).check(doesNotExist());
-    }
+    /**
+     * Helper function to create a notification from the event index and target channel.
+     * @param idx The index of the event for the notification
+     * @param channel The channel of the communication
+     * @param offset The notification number, to allow for multiple notifications for events with
+     *               the same channel.
+     * @return a notification for the selected event.
+     */
+    private static Notification templateNotification(
+            Integer idx, Notification.Channel channel, Integer offset) {
```

Comment:
I don't see anywhere that uses any offset other than zero. So that can be removed until proven necessary. When it is proven necessary, it should be added with a default of zero.

Otherwise, it looks good. Good job!